src/vsr/replica.zig

```diff
@@ -7734,6 +7788,43 @@ pub fn ReplicaType(
             self.sync_dispatch(.idle);
         }
 
+        /// We have just:
+        /// - finished superblock sync,
+        /// - replaced our superblock,
+        /// - repaired the manifest blocks,
+        /// - and opened the state machine.
+        /// Now we sync:
+        /// - the missed LSM table blocks (index/filter/data), and
+        /// - client reply messages.
+        fn sync_content(self: *Self) void {
+            assert(self.syncing == .idle);
+            assert(self.state_machine_opened);
+            assert(self.superblock.working.vsr_state.sync_op_max > 0);
+
+            for (0..constants.clients_max) |entry_slot| {
+                if (self.superblock.client_sessions.entries_free.isSet(entry_slot)) {
+                    assert(!self.client_replies.faulty.isSet(entry_slot));
+                } else {
+                    const entry = &self.superblock.client_sessions.entries[entry_slot];
+                    if (entry.header.op >= self.superblock.working.vsr_state.sync_op_min and
+                        entry.header.op <= self.superblock.working.vsr_state.sync_op_max)
+                    {
+                        const entry_faulty = entry.header.size > @sizeOf(Header);
+                        self.client_replies.faulty.setValue(entry_slot, entry_faulty);
+                    }
+                }
+            }
+        }
+
+        pub fn sync_content_done(self: *const Self) bool {
+            if (self.superblock.staging.vsr_state.sync_op_max == 0) {
+                return true;
+            } else {
+                // TODO Return whether all tables are synced.
+                return self.client_replies.faulty.count() == 0;
```

Comment:
:thinking: we could have faulty replies outside of the min-max range. So perhaps this should `.iterate` and check if the bit is in range?